Component hierarchy and refactor

dentemple opened the pull request:
The purpose of this PR is to scaffold the front page's components to be closer to the view from our component design document.

The intent is to, hopefully, allow people to drop in and build up minor pieces quickly and asynchronously.

- This PR closes Issue #4 
- Separates the larger Event Listing and Event List components into smaller components
- Scaffolds additional components for the App Header
- Scaffolds an additional page layer between Events and App called Main
- Any css that could be made inline is made inline; Radium is still being imported for situations that can't be covered by inline-css (such as with active links)

(My apologies if the code is a little sloppy.  Feel free to throw this back on to me with any suggested changes.)

Thank you!

aaghevli commented on src/components/MainHeader.js:
```diff
@@ -0,0 +1,29 @@
+import React, { Component, PropTypes } from 'react';
+import Radium, { Style } from 'radium';
+
+import DayPanel from './DayPanel'
```
Its a bit unclear what 'DayPanel' means. Is there a slightly more abstract name like ListHeader or SummaryPanel that does describes the position vs the contents?

aaghevli commented on src/components/SearchInput.js:
```diff
@@ -0,0 +1,37 @@
+      color: '#222',
+      outline: 'none',
+      background: 'url(../../../src/assets/img/search-icon.png) no-repeat 95% center',
+      // '::placeholder': {
```
nit: remove comments

aaghevli commented on src/containers/Main.js:
```diff
@@ -0,0 +1,52 @@
+class Main extends Component {
+  constructor(props) {
+    super(props);
+//   this.state = {
```
nit: comments

dentemple commented:
- ListHeader definitely makes more sense as a name.  I renamed the component (including its nested components).
- Nixed the placeholder comments
- Fixed an issue with the relative links, which got broken when I moved them
